url1d5e9d990f_Telsis_jOCP_tgz-pJ8-com_telsis_jocp_sampleApp_BasicOCPAppJ8/src/com/telsis/jocp/messages/TelsisHandlerResult.java

```diff
@@ -0,0 +1,178 @@
+ * <td>64</td>
+ * <td>Custom data</td>
+ * </tr>
+ * </table>
+ *
+ * @see TelsisHandler
+ * @see TelsisHandlerWithParty
+ * @author Telsis
+ */
+public class TelsisHandlerResult extends CallControlMessage {
+    /** The message type. */
+    public static final LegacyOCPMessageTypes TYPE = LegacyOCPMessageTypes.TELSIS_HANDLER_RESULT;
+    /** The expected length of this message. */
+    private static final int            EXPECTED_LENGTH = 66;
+
+    /** The length of the returnData field. */
+    private static final int            RETURN_DATA_LENGTH = 64;
+
+    /** The index into the zip table on the SCP for this result. */
+    private byte                        zipNumber;
+    /** The return data. */
+    private final byte[]                returnData = new byte[RETURN_DATA_LENGTH];
+
+    /**
+     * Decode the buffer into a Telsis Handler Result message.
+     *
+     * @param buffer
+     *            the message to decode
+     * @throws OCPException
+     *             if the buffer could not be decoded
+     */
+    public TelsisHandlerResult(final ByteBuffer buffer) throws OCPException {
+        super(buffer);
+        super.advance(buffer);
+
+        if (buffer.limit() != EXPECTED_LENGTH) {
+            throw new CallMessageException(
+                    getDestTID(),
+                    getOrigTID(),
+                    TYPE.getCommandCode(),
+                    CallCommandUnsupported.REASON_LENGTH_UNSUPPORTED,
+                    (short) buffer.limit());
+        }
+
+        buffer.get(); // for word alignment
+        zipNumber = buffer.get();
+        buffer.get(returnData);
+    }
+
+    /**
+     * Instantiates a new Telsis Handler Result message.
+     */
+    public TelsisHandlerResult() {
+        super(TYPE.getCommandCode());
+    }
+
+    @Override
+    protected final void encode(final ByteBuffer buffer) {
+        super.encode(buffer);
+        buffer.put((byte) 0); // for word alignment
+        buffer.put(zipNumber);
+        buffer.put(returnData);
+    }
+
+    /**
+     * Gets the zip number.
+     *
+     * @return the zip number
+     */
+    public final byte getZipNumber() {
+        return zipNumber;
+    }
+
+    /**
+     * Sets the zip number.
+     *
+     * @param newZipNumber the new zip number
+     */
+    public final void setZipNumber(final byte newZipNumber) {
+        this.zipNumber = newZipNumber;
+    }
+
+    /**
+     * Gets the return data decoded as a TelsisHandlerPayload instance.
+     *
+     * @param <T>
+     *            Type parameter for the payload class
+     * @param payloadClass
+     *            The class of the payload to decode
+     * @return the decoded payload
+     */
+    public final <T extends TelsisHandlerPayload> T getPayload(
+            final Class<T> payloadClass) {
+        try {
+            T payload = payloadClass.newInstance();
+            payload.decode(ByteBuffer.wrap(returnData));
+            return payload;
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)